### ISSUE TITLE
Do not load full source into RAM on write_to_deltalake

### ISSUE THREAD

**Issue by aersam** [delta-io/delta-rs#2255]
# Description
In [python/lib.rs](https://github.com/delta-io/delta-rs/blob/main/python/src/lib.rs#L1404), the first thing that happens on `write_to_deltalake` is to collect to batches to a Vec. This loads all RecordBatches into RAM, no? This seems like not a good thing to me. I think the main reason is that write.rs tries to get the schema from the batches, but the schema would have been known in python anyway, so why not pass it directly?

**Use Case**
I don't want to waste resources ;)

**Related Issue(s)**

**Comment by ion-elgreco** [comment]
@aersam correct, it's not the efficient way to do that :) Will already mentioned an improvement over that, which I've logged here https://github.com/delta-io/delta-rs/issues/1984

**Comment by aersam** [comment]
I can pick it up, but I'd rather do it on the write.rs operation

**Comment by ion-elgreco** [comment]
@aersam that's fine!

**Comment by aersam** [comment]
Ok, I see partioning makes this quite complicated 🙂 And MemoryExec of DataFusion is not helpful, so might take some time
